src/stubs/win32/opamWindows.c

```diff
@@ -549,24 +551,29 @@ CAMLprim value OPAMW_process_putenv(value pid, value key, value val)
     caml_failwith(result);
 }
 
-/*
- * Somewhat against my better judgement, wrap SHGetFolderPath rather than
- * SHGetKnownFolderPath to maintain XP compatibility. OPAM already requires
- * Windows Vista+ because of GetCurrentConsoleFontEx, but there may be a
- * workaround for that for XP lusers.
- */
-CAMLprim value OPAMW_SHGetFolderPath(value nFolder, value dwFlags)
+value OPAMW_SHGetKnownFolderPath(REFKNOWNFOLDERID rfid)
 {
-  WCHAR szPath[MAX_PATH];
-
-  if (SUCCEEDED(SHGetFolderPath(NULL,
-                                Int_val(nFolder),
-                                NULL,
-                                Int_val(dwFlags),
-                                szPath)))
-    return caml_copy_string_of_utf16(szPath);
-  else
-    caml_failwith("OPAMW_SHGetFolderPath");
+  PWSTR path = NULL;
+  value result;
+
+  if (SUCCEEDED(SHGetKnownFolderPath(rfid, (DWORD)0, NULL, &path))) {
+    result = caml_copy_string_of_utf16(path);
+    CoTaskMemFree((LPVOID)path);
+    return result;
+  } else {
+    CoTaskMemFree((LPVOID)path);
```

Comment:
Entirely under style/choice, but the casts are just noise:
```suggestion
  if (SUCCEEDED(SHGetKnownFolderPath(rfid, 0, NULL, &path))) {
    result = caml_copy_string_of_utf16(path);
    CoTaskMemFree(path);
    return result;
  } else {
    CoTaskMemFree(path);
```